tests/integration/Payment/UseCases/PaymentCompletedEmailTest.php

```diff
@@ -98,86 +94,58 @@ public function testWhenGroupHasNoMailCredentialsSetNothingHappens() : void
         $this->paymentService->completePayment(1);
 
         $this->assertPaymentWasCompleted();
-        $this->tester->seeEmailCount(0);
     }
 
     public function testEmailIsSentWhenPaymentIsCompleted() : void
```

Comment:
Tady teď úplně chybý ten výsledný assert. Takže e2e test, který má kontrolovat, že při dokončení platby se odešle mail, nic nekontroluje.

Reply:
to nevim jak bys chtel delat bez toho catchera. Muzu smazat ten test

Reply:
Stačí použít nějakej hloupej mailer, kterej bude jen počítat maily:

```php
final class FakeMailer implements IMailer
{
    private int $sentMessagesCount = 0;

    public function send(Message $message) : void
    {
        $this->sentMessagesCount++;
    }

    public function getSentMessagesCount() : int
    {
        return $this->sentMessagesCount;
    }
}
```
to, že se změnila jedna část aplikace přece neznamená, že smažeme všechny testy, který se jí nějak dotýkají :grinning:

Reply:
ja prave ale nevim, jak se dostanu k tomu `getSentMessagesCount` protoze tam s tim objektem nikde nepracujeme, ten je jen v configu a neni nikde vytazeny. Takze si ho grapnout?